### PR TITLE
build: Apply webui lint fixes in precommit

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This updates the web linters to automatically apply fixes when doing a
pre-commit check. This should ideally streamline the commit process to
reduce the amount of times the user needs to run prettier and eslint
before committing.

## Test Plan

When committing code, if the code contains issues that a linter can fix automatically, those fixes should be applied ~while also preventing the commit from going through~.
things to check:

* prettier: try adding a statement to a ts file without a semicolon
* eslint: try disabling a test in a ts file by replacing the `it` call with `xit`
* stylelint: try adding a blank line in between two style declaration blocks

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
[WEB-928]

[WEB-928]: https://determinedai.atlassian.net/browse/WEB-928?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by keita-determined** [comment]
the change doesn't stage the file(s) after the fix.

**Comment by ashtonG** [comment]
It shouldn't! The idea is to prompt the user to inspect the fixes and proceed from there.

**Comment by keita-determined** [comment]
the jira ticket says
>The pre-commit should also re-add the files to git and finish the commit if all issues were addressed by running the commands (eslint errors that cannot be automatically fixed should still block commit).

so we should support this?

**Comment by ashtonG** [comment]
there isn't a way to do this such that it only affects webui and not the rest of the codebase, AFAICT.

File: pre-commit/web_lint_check.py [hunk 1]
```diff
@@ -22,18 +22,19 @@ def web_lint_check():
...
-        run_command += ["eslint", f"ES_ARGS={file_paths}"]
+        run_command += ["eslint", f"ES_ARGS=-- --fix {file_paths}"]
     elif target == "css":
-        run_command += ["stylelint", f"ST_ARGS={file_paths}"]
+        run_command += ["stylelint", f"ST_ARGS=-- --fix {file_paths}"]
     elif target == "misc":
         run_command += ["check-package-lock"]
 
     returncode: int = subprocess.call(run_command)
+    subprocess.call(["git", "add"] + args.file_paths)
```
Comment (keita-determined) [inline pre-commit/web_lint_check.py:37]: `args.file_paths` is `Any` type, so could you add a type like the following

```diff
diff --git a/pre-commit/web_lint_check.py b/pre-commit/web_lint_check.py
index 5c8197b2f..a3c36df7f 100755
--- a/pre-commit/web_lint_check.py
+++ b/pre-commit/web_lint_check.py
@@ -4,6 +4,7 @@ import argparse
 import multiprocessing
 import os
 import subprocess
+from typing import List
 
 
 def web_lint_check():
@@ -14,7 +15,8 @@ def web_lint_check():
     DIR = "webui/react/"
 
     target: str = args.target
-    file_paths: str = " ".join([os.path.relpath(file_path, DIR) for file_path in args.file_paths])
+    args_file_paths: List[str] = args.file_paths
+    file_paths: str = " ".join([os.path.relpath(file_path, DIR) for file_path in args_file_paths])
     nproc: int = multiprocessing.cpu_count()
     run_command: list[str] = [
         "make",
@@ -34,7 +36,7 @@ def web_lint_check():
         run_command += ["check-package-lock"]
 
     returncode: int = subprocess.call(run_command)
-    subprocess.call(["git", "add"] + args.file_paths)
+    subprocess.call(["git", "add"] + args_file_paths)
     exit(returncode)
```

**Review by keita-determined — APPROVED** [review]
requested 1 change, otherwise LGTM

File: pre-commit/web_lint_check.py (hunk 1 above)
Comment (keita-determined) [reply pre-commit/web_lint_check.py:37]: actually, we can do `parser.add_argument("file_paths", help="1 or more file paths", nargs="+", type=List[str])` too, but type wasnt correctly assigned on my local

Comment (ashtonG) [reply pre-commit/web_lint_check.py:37]: the second method doesn't seem to communicate the type to the type checker, does it?

**Pull request merged**